src/core/selection/selection.ts

```diff
@@ -6,5 +6,11 @@
  * ---------------------------------------------------------------------
  */
 import { SchemaElement } from '../model';
+import { LinkedUISchemaElement } from '../model/uischema';
 
-export type SelectedElement = SchemaElement | undefined;
+export type SelectedElement = WrappedSelectedElement | undefined;
+
+interface WrappedSelectedElement {
+  uiSchema: LinkedUISchemaElement;
+  schema?: SchemaElement;
+}
```

Comment:
Why do we need the `WrappedSelectedElement`, couldn't we just use `LinkedUISchemaElement`? When we need the corresponding schema element we can resolve it.

Reply:
When we set the selection, we already have the schema element. So there's no need to search for the corresponding schema element each time the selection changes. Depending on the size of the schema, that could slow things down a bit. But since we don't work with large schemas yet, I don't have a hard opinion on this.

Reply:
But the `LinkedUISchemaElement` has a list of `linkedSchemaElements`. Can't we use it?

Reply:
the `linkedSchemaElements` contains only the uuid of the schema elements, so we would have to search for the correct element in the schema.

Reply:
why can't we link the full schema? the `LinkedUISchemaElement` is not serialized is it?

Reply:
The LinkedUISchemaElements and SchemaElements are not stable, we would need to replace the instances from the `linkedSchemaElements` and `linkedUISchemaElements` on each re-render. It would also make the cloning more complex, since the schemas would not be trees anymore.

Reply:
I see, and we can't have a map to easier retrieve what we need?